### PR TITLE
✨ Different colors for labels

### PR DESCRIPTION
**Pull request by ken479**
# Description

Assign different colors for respective labels

differentcolorbndbox

differentcolorpolygon

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Tested on?

- [x] Windows  
- [ ] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: src/shared/services/label-color.services.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+@Injectable({
+    providedIn: 'root',
+})
+export class LabelColorServices {
+    private readonly fixedColors: string[];
+    index!: number;
+    labelColorList!: Map<string, string>;
+
+    constructor() {
```
Comment (devenyantis) [inline src/shared/services/label-color.services.ts:11]: Looks like it is limited to only 20 colors. What happen after that? Is there any other way to get color. maybe using color picker library

Comment (ken479) [reply src/shared/services/label-color.services.ts:11]: @devenyantis Hi Deven, the first 20 colors of the labels stay similar to all the projects, 21st color onwards will be randomly generated using the function hue saturation and lightness below there. Previously I did using only randomly pick colors, I find that every time reenters the project, all label colors will randomly pick again, so I am afraid users may confused. That's why I coded this way.

Comment (devenyantis) [reply src/shared/services/label-color.services.ts:11]: I think it doesn't matter. as long as there are color differences. But, anyway, no harm to go as you build as well.
Thanks

**Review by devenyantis — APPROVED** [review]

**Comment by devenyantis** [comment]
Can you solve the branch conflicts first @ken479 
Thanks

File: src/shared/services/label-color.services.ts (hunk 1 above)
Comment (ken479) [reply src/shared/services/label-color.services.ts:11]: Thanks @devenyantis

**Comment by ken479** [comment]
> Can you solve the branch conflicts first @ken479 Thanks

@devenyantis sorry about that, conflict is resolved

**Pull request merged**